src/services/PingPongService.ts

```diff
@@ -1,62 +1,50 @@
-import { geoToH3 } from 'h3-js';
 import { utils } from 'ethers';
-
 import { eip712, utils as vUtils } from '@windingtree/videre-sdk';
-import { constants } from '@windingtree/videre-sdk/dist/cjs/utils';
-
-import facilityRepository from '../repositories/FacilityRepository';
-
 import log from './LogService';
 import WakuService from './WakuService';
 import WalletService from './WalletService';
-
 import { walletAccountsIndexes } from '../types';
-import { Facility } from '../proto/facility';
 import { Ping, Pong } from '../proto/pingpong';
-import { LatLng } from '../proto/latlng';
-
 import { typedDataDomain, videreConfig } from '../config';
-import ApiError from '../exceptions/ApiError';
 
-export default class PingPongService {
-  protected waku: WakuService;
-  protected unsubscribe: () => void = () => {
-    return;
-  };
+const unsubscribeHandler: () => void = () => {
+  return;
+};
 
-  /**
-   * After database initialization, start the ping / pong service
-   * for a specified facility.
-   */
-  public async start(facilityId: string): Promise<void> {
-    const metadata = await facilityRepository.getFacilityKey<Facility>(
-      facilityId,
-      'metadata'
-    );
+interface Unsubscribe {
+  h3Index: string;
+  handler: typeof unsubscribeHandler;
+}
 
-    if (metadata === null) {
-      throw ApiError.NotFound(
-        `Unable to find "metadata" of the facility: ${facilityId}`
-      );
-    }
+export class PingPongService {
+  protected waku: WakuService;
 
-    const loc = metadata.location;
+  protected unsubscribes = new Map<string, Unsubscribe>();
+  public locsManaged = new Map<string, string[]>();
+  public facilityToLoc = new Map<string, string>();
 
+  /**
+   * Start the ping / pong service for a specified facility.
+   */
+  public async start(facilityId: string, h3Index: string): Promise<void> {
     if (!this.waku) this.waku = WakuService.getInstance();
 
-    if (loc) {
-      // get the h3 index to monitor from the location
-      const h3Index = geoToH3(
-        loc.latitude,
-        loc.longitude,
-        constants.DefaultH3Resolution
-      );
+    //store Maps
+    this.facilityToLoc.set(facilityId, h3Index);
+    if (this.locsManaged.has(h3Index)) {
+      const locs = this.locsManaged.get(h3Index);
+      locs?.push(facilityId);
+      this.locsManaged.set(h3Index, locs || []);
+    } else {
+      this.locsManaged.set(h3Index, [facilityId]);
+    }
 
-      // watch for pings
-      this.unsubscribe = await this.waku.makeWakuObserver(
-        async (message) => {
-          const msg = this.waku.processMessage(Ping, message);
-          if (msg) log.green(`Ping received with timestamp: ${msg.timestamp}`);
+    // watch for pings
+    const observer = await this.waku.makeWakuObserver(
+      async (message) => {
+        const msg = this.waku.processMessage(Ping, message);
+        if (msg) {
+          log.green(`Ping received with timestamp: ${msg.timestamp}`);
```

Comment:
`msg.timestamp` needs to be guarded here as this may be null if not specified by the sender.

Reply:
I'd suggest still providing a respond, but perhaps log the timestamp as the time **received** if there is no `msg.timestamp` that was passed in the message.